src/auth/jwt.js

```diff
@@ -1,17 +1,134 @@
+import { userService, logger } from '../di-container'
+import config from '../libs/config'
+import { generateTokenCookieOptions } from '../libs/cookies'
+import InvalidTokenError from '../errors/InvalidTokenError'
+import ForbiddenError from '../errors/ForbiddenError'
+
+const ACCESS_TOKEN_TTL = config.get('security:jwt:access_token_ttl')
+
+const attachJwtPayloadToRequest = (req, { requestProperty='auth', payload, fieldsToInclude=[] }) => {
+    req[requestProperty] = {}
+    fieldsToInclude.forEach(fieldname => {
+      return req[requestProperty][fieldname] = payload[fieldname]
+    })
+  }
 
 /**
  * Generates JWT from user object
  * @param {object} user User object
  * @param {int|string} expiresIn expressed in seconds or a string describing a time span zeit/ms. Eg: 60, "2 days", "10h", "7d"
  * @returns {string} Standard JWT payload string 
  */
-export function generateJWT(user, expiresIn = null){
-    return sign({
+export async function generateJWT(user, expiresIn = ACCESS_TOKEN_TTL){
+    return jwtSign({
         id: user.id
-    }, process.env.JWT_SECRET, {
-        expiresIn: expiresIn === null ? DEFAULT_JWT_TTL : expiresIn 
+    }, config.get('security:jwt:secret'), {
+        expiresIn
     })
+}
+
+/**
+ * Generates JWT expirty date
+ * @param {int} expiryTtl Number of milliseconds from current time
+ * @returns {Date} Calculated expiry datetime
+ */
+export const generateJwtExpiryDate = expiryTtl => new Date(new Date().getTime() + parseInt(expiryTtl))
+
+/**
+* Middleware for validating jwt token to protect protected routes
+* @param {Object} req Request object
+* @param {Object} res Response object
+* @param {Function} next next function
+*/
+export const validateJwt = async (req, res, next) => {
+    if(!req || (req && !req.cookies)) {
+        logger.log(logger.LOGLEVEL.ERROR, { logMessage: 'Missing cookies' })
+        next(new ForbiddenError('Missing cookies'))
+        return
+    }
+
+    if(!req.cookies.accessToken || !req.cookies.refreshToken) {
+        logger.log(logger.LOGLEVEL.ERROR, { logMessage: 'Missing access or refresh token' })
+        next(new ForbiddenError('Missing access or refresh token'))
+        return
+    }
+    const accessToken = req.cookies.accessToken
+    const refreshToken = req.cookies.refreshToken
+
+    try {
+        const decodedJwtPayload = await jwtVerify(accessToken, config.get('security:jwt:secret'))
+        attachJwtPayloadToRequest(req, {
+            payload: decodedJwtPayload,
+            fieldsToInclude: [ 'id' ]
+        })
+        // next(null, user)
+        next()
+    } catch(err) {
+        // access token expired; issue new access token using refresh token
+        if(err.name === 'TokenExpiredError') {
+            let user = null
+
+            try {
+                // extract user id from refresh token
+                const { id: uid} = await jwtVerify(refreshToken, config.get('security:jwt:secret'))
```

Comment:
In theory here we could search by refresh token instead and load the associated user if anyone has one?

Reply:
@ryanolee Ok.  Thanks for the suggestion.  Will do that in the next PR push.